mrfilter: fixed gradient stdev bug #642.

maxpietsch opened the pull request:
…ot set if stdev.size() == 3

maxpietsch commented:
Any objections against merging this?

thijsdhollander commented:
Didn't have time to test it, but it seems the logic for the change is sound...?  So happy to merge if you're confident that it works. ;-)

maxpietsch commented:
Wasn't confident so I did some more testing. I found another bug: If -magnitude is set then the gradient filter defaults to gradients wrt scanner space, without -magnitude it defaults to the opposite. (The default in `mrfilter` is not wrt scanner space).

There is a warning in gradient.h 

   if (do_wrt_scanner && magnitude)
       WARN(For a gradient magnitude image, setting gradient to scanner axes has no effect)

which I think is not necessary. I don't see why a gradient magnitude shouldn't be computed wrt to scanner space. 

So I removed the above warning and changed the behaviour of -magnitude to the one set via `.compute_wrt_scanner()` in fc0b6a5.

Also, I wrote some tests for `mrfilter gradient`. I noticed that there are 4 files (mrfilter/out8-12.mif) that are not used for any test. Does anyone use those or shall I get rid of them?
